// Copyright (c) 2010-2021 The Open-Transactions developers
// This Source Code Form is subject to the terms of the Mozilla Public
// License, v. 2.0. If a copy of the MPL was not distributed with this
// file, You can obtain one at http://mozilla.org/MPL/2.0/.

#include "opentxs/protobuf/verify/ServerRequest.hpp"  // IWYU pragma: associated

#include "opentxs/protobuf/Basic.hpp"
#include "opentxs/protobuf/Enums.pb.h"
#include "opentxs/protobuf/OTXEnums.pb.h"
#include "opentxs/protobuf/ServerRequest.pb.h"
#include "opentxs/protobuf/verify/Nym.hpp"              // IWYU pragma: keep
#include "opentxs/protobuf/verify/Signature.hpp"        // IWYU pragma: keep
#include "opentxs/protobuf/verify/StorageItemHash.hpp"  // IWYU pragma: keep
#include "opentxs/protobuf/verify/VerifyOTX.hpp"
#include "protobuf/Check.hpp"

#define PROTO_NAME "server request"

namespace opentxs::proto
{
auto CheckProto_1(const ServerRequest& input, const bool silent) -> bool
{
    CHECK_IDENTIFIER(id);
    CHECK_IDENTIFIER(nym);
    CHECK_IDENTIFIER(server);
    CHECK_SUBOBJECT_VA(
        signature, ServerRequestAllowedSignature(), SIGROLE_SERVERREQUEST);

    switch (input.type()) {
        case SERVERREQUEST_ACTIVATE: {
        } break;
        case SERVERREQUEST_ERROR:
        default: {
            FAIL_1("Invalid type");
        }
    }

    OPTIONAL_SUBOBJECT(credentials, ServerRequestAllowedNym());

    return true;
}

auto CheckProto_2(const ServerRequest& input, const bool silent) -> bool
{
    return CheckProto_1(input, silent);
}

auto CheckProto_3(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(3)
}

auto CheckProto_4(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(4)
}

auto CheckProto_5(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(5)
}

auto CheckProto_6(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(6)
}

auto CheckProto_7(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(7)
}

auto CheckProto_8(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(8)
}

auto CheckProto_9(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(9)
}

auto CheckProto_10(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(10)
}

auto CheckProto_11(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(11)
}

auto CheckProto_12(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(12)
}

auto CheckProto_13(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(13)
}

auto CheckProto_14(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(14)
}

auto CheckProto_15(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(15)
}

auto CheckProto_16(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(16)
}

auto CheckProto_17(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(17)
}

auto CheckProto_18(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(18)
}

auto CheckProto_19(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(19)
}

auto CheckProto_20(const ServerRequest& input, const bool silent) -> bool
{
    UNDEFINED_VERSION(20)
}
}  // namespace opentxs::proto
